client/src/main/java/com/netflix/conductor/client/task/WorkflowTaskCoordinator.java

```diff
@@ -297,32 +305,43 @@ private void pollForTask(Worker worker) {
 			logger.error("Error when polling for task " + e.getMessage(), e);
 		}
 	}
-	
-	private void execute(Worker worker, Task task) {
+
+	private void handleException(Throwable t, TaskResult result, Worker worker, boolean updateTask, Task task) {
+		WorkflowTaskMetrics.executionException(worker.getTaskDefName(), t);
+		result.setStatus(TaskResult.Status.FAILED);
+		result.setReasonForIncompletion("Error while executing the task: " + t);
+		TaskExecLog execLog = result.getLog();
+		execLog.setError(t.getMessage());
+		for (StackTraceElement ste : t.getStackTrace()) {
+            execLog.getErrorTrace().add(ste.toString());
+        }
+		updateWithRetry(updateRetryCount, task, result, worker);
+	}
+
+	private TaskResult execute(Worker worker, Task task) {
```

Comment:
Why did you change this to return TaskResult? You're still handling the result within the body and you ignore it in the onSuccess above.